src/Interpreters/inplaceBlockConversions.cpp

```diff
@@ -173,6 +190,53 @@ std::optional<ActionsDAG> createExpressions(
     return ActionsDAG::merge(std::move(dag), std::move(actions));
 }
 
+std::optional<ActionsDAG> createExpressionsAnalyzer(
+    const Block & header,
+    ASTPtr expr_list,
+    bool save_unneeded_columns,
+    ContextPtr context)
+{
+    if (!expr_list)
+        return {};
+
+    auto execution_context = Context::createCopy(context);
+    auto expression = buildQueryTree(expr_list, execution_context);
+
+    ColumnsDescription fake_column_descriptions(header.getNamesAndTypesList());
+    auto storage = std::make_shared<StorageDummy>(StorageID{"dummy", "dummy"}, fake_column_descriptions);
+    QueryTreeNodePtr fake_table_expression = std::make_shared<TableNode>(storage, execution_context);
+
+    QueryAnalyzer analyzer(false);
+    analyzer.resolveConstantExpression(expression, fake_table_expression, execution_context);
```

Comment:
```suggestion
    analyzer.resolve(expression, fake_table_expression, execution_context);
```
Because it's not a constant expression